src/mscorlib/src/System/Runtime/Intrinsics/X86/Sse41.cs

```diff
@@ -165,37 +214,86 @@ public static class Sse41
+            return InsertByte<T>(value, data, index);
+        }
 
         /// <summary>
         /// __m128i _mm_insert_epi32 (__m128i a, int i, const int imm8)
         /// </summary>
-        public static Vector128<T> InsertInt<T>(Vector128<T> value, int data, byte index) where T : struct => InsertInt<T>(value, data, index);
+        public static Vector128<T> InsertInt<T>(Vector128<T> value, int data, byte index) where T : struct
+        {
+            if (IntrinsicUtil.IsNonNumericType<T>())
+            {
+                throw new NotSupportedException(SR.Arg_TypeNotSupported); 
+            }
+            return InsertInt<T>(value, data, index);
+        }
 
         /// <summary>
         /// __m128i _mm_insert_epi32 (__m128i a, int i, const int imm8)
         /// </summary>
-        public static Vector128<T> InsertUint<T>(Vector128<T> value, uint data, byte index) where T : struct => InsertUint<T>(value, data, index);
+        public static Vector128<T> InsertUint<T>(Vector128<T> value, uint data, byte index) where T : struct
+        {
+            if (IntrinsicUtil.IsNonNumericType<T>())
+            {
+                throw new NotSupportedException(SR.Arg_TypeNotSupported); 
+            }
+            return InsertUint<T>(value, data, index);
+        }
 
         /// <summary>
         /// __m128i _mm_insert_epi64 (__m128i a, __int64 i, const int imm8)
         /// </summary>
-        public static Vector128<T> InsertLong<T>(Vector128<T> value, long data, byte index) where T : struct => InsertLong<T>(value, data, index);
+        public static Vector128<T> InsertLong<T>(Vector128<T> value, long data, byte index) where T : struct
+        {
+            if (IntrinsicUtil.IsNonNumericType<T>())
+            {
+                throw new NotSupportedException(SR.Arg_TypeNotSupported); 
+            }
+            return InsertLong<T>(value, data, index);
+        }
 
         /// <summary>
         /// __m128i _mm_insert_epi64 (__m128i a, __int64 i, const int imm8)
         /// </summary>
-        public static Vector128<T> InsertUlong<T>(Vector128<T> value, ulong data, byte index) where T : struct => InsertUlong<T>(value, data, index);
+        public static Vector128<T> InsertUlong<T>(Vector128<T> value, ulong data, byte index) where T : struct
+        {
+            if (IntrinsicUtil.IsNonNumericType<T>())
+            {
+                throw new NotSupportedException(SR.Arg_TypeNotSupported); 
+            }
+            return InsertUlong<T>(value, data, index);
+        }
 
         /// <summary>
         /// __m128 _mm_insert_ps (__m128 a, __m128 b, const int imm8)
         /// </summary>
-        public static Vector128<T> InsertFloat<T>(Vector128<T> value, float data, byte index) where T : struct => InsertFloat<T>(value, data, index);
+        public static Vector128<T> InsertFloat<T>(Vector128<T> value, float data, byte index) where T : struct
+        {
+            if (IntrinsicUtil.IsNonNumericType<T>())
```

Comment:
You can wrap the whole `if (...) throw ...;` into a method to minimize the repeated pattern. A good place to put the helper method may be `mscorlib\src\System\ThrowHelper.cs` - it is our dumping ground for these sort of throw methods.

Reply:
Good point! Will do.